### PR TITLE
[NCL-6190] implement analyzer of deliverables endpoint

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

Draft, still need to create some tests

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            log.error("Error trying to start analysis of deliverables task for milestone: {}", milestoneId, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void addMDCHeaders(Set<Request.Header> headers) {
+        headersFromMdc(headers, MDCHeaderKeys.REQUEST_CONTEXT);
+        headersFromMdc(headers, MDCHeaderKeys.PROCESS_CONTEXT);
+        headersFromMdc(headers, MDCHeaderKeys.TMP);
+        headersFromMdc(headers, MDCHeaderKeys.EXP);
```
Comment (matejonnet) [inline facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java:91]: I guess tmp and exp have no meaning in this case.
Where is the PROCESS_CONTEXT set ?

File: bpm/src/main/java/org/jboss/pnc/bpm/task/AnalyzeDeliverablesTask.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.jboss.pnc.api.dto.Request;
+import org.jboss.pnc.bpm.BpmTask;
+import org.jboss.pnc.bpm.model.AnalyzeDeliverablesBpmRequest;
+import org.jboss.pnc.spi.exception.CoreException;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+@ToString(callSuper = true)
+public class AnalyzeDeliverablesTask extends BpmTask {
```
Comment (matejonnet) [inline bpm/src/main/java/org/jboss/pnc/bpm/task/AnalyzeDeliverablesTask.java:14]: What do we get by extending BpmTask for a stateless approach and only RestConnector support ?

**Review by matejonnet — CHANGES_REQUESTED** [review]

File: facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java (hunk 1 above)
Comment (michalovjan) [reply facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java:91]: These headers are for the callback that BPM will invoke once DelAnal is done. The other MDC parameters which we discussed to duplicate, have in its own class and are directly for BPM, are set here https://github.com/project-ncl/pnc/pull/3461/files#diff-d9880356c1a2790e5f7caab27c18a69df6cb556a67c2f00631aa9aa4c458b193R83

File: bpm/src/main/java/org/jboss/pnc/bpm/task/AnalyzeDeliverablesTask.java (hunk 2 above)
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/task/AnalyzeDeliverablesTask.java:14]: The only thing I needed from BpmTask.class is the method getExtendedProcessParamameters() which returns the actual parameters.

https://github.com/project-ncl/pnc/blob/e05d714bd0df726ad579ad19c4073ec5eb501c22/bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java#L168-L200

**Comment by michalovjan** [comment]
retest this please

File: facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java (hunk 1 above)
Comment (michalovjan) [reply facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java]: It is set now in PMEndpoint

File: bpm/src/main/java/org/jboss/pnc/bpm/task/AnalyzeDeliverablesTask.java (hunk 2 above)
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/task/AnalyzeDeliverablesTask.java]: Now the task doesn't use BpmTask. We get processParameters through direct json serialization.

File: bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import org.jboss.pnc.bpm.model.BpmEvent;
+import org.jboss.pnc.bpm.model.MDCParameters;
```
Comment (matejonnet) [inline bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java:25]: ?

File: facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import java.util.Set;
+
+@ApplicationScoped
+public class DeliverablesAnalyzerInvokerImpl implements DeliverablesAnalyzerInvoker {
+
+    private static final Logger log = LoggerFactory.getLogger(DeliverablesAnalyzerInvokerImpl.class);
+
+    private UserService userService;
+
+    private BpmManager bpmManager;
```
Comment (matejonnet) [inline facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java:51]: ?

**Review by matejonnet — CHANGES_REQUESTED** [review]
Some clean-up required otherwise it looks good to me.

File: bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java (hunk 3 above)
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java]: yes indeed... ????

File: facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java (hunk 4 above)
Comment (michalovjan) [reply facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java]: cleaned!

**Pull request merged**